covalent_awslambda_plugin/awslambda.py

```diff
@@ -492,14 +287,10 @@ async def run(self, function: Callable, args: List, kwargs: Dict, task_metadata:
         """
         dispatch_id = task_metadata["dispatch_id"]
         node_id = task_metadata["node_id"]
-        workdir = os.path.join(self._cwd, dispatch_id)
+        workdir = self.cache_dir
 
         func_filename = FUNC_FILENAME.format(dispatch_id=dispatch_id, node_id=node_id)
         result_filename = RESULT_FILENAME.format(dispatch_id=dispatch_id, node_id=node_id)
-        lambda_function_name = LAMBDA_FUNCTION_NAME.format(
-            dispatch_id=dispatch_id, node_id=node_id
-        )
-
         app_log.debug(f"In run for task - {dispatch_id} - {node_id} ... ")
 
         app_log.debug("Pickling function, args and kwargs..")
```

Comment:
This should be moved to `_upload_task_sync` so as to avoid blocking the main thread, but you or @FyzHsn can make this change later to close #45 .

Reply:
@venkatBala Just wanted to check up on the status of this change - should I make this update or is this done already?

Reply:
@FyzHsn could you please make that change? Just wanted to avoid scope creep : )